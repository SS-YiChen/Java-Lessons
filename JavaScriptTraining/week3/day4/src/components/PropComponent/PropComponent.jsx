import { useState } from "react";

// I can be told what to display using props
export const PropComponent = (props) => {
    // Instead of choosing via state
    // const [name, setName] = useState('');
    // const [age, setAge] = useState(0);
    // const [bio, setBio] = useState('');
    
    return (
        <>
            <h1>Name: {props.name}</h1>
            <h2>Age: {props.age}</h2>
            {props.children}
        </>
    );   
}

export default PropComponent;