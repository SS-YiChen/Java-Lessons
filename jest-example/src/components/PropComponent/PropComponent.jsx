import { useState } from "react";

export const PropComponent = (props) => {

    return (
        <>
            <h1>Name: {props.name}</h1>
            <h2>Age: {props.age}</h2>
            {props.children}
        </>
    );
}

export default PropComponent;