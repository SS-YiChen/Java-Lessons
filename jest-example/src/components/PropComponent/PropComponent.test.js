import renderer from 'react-test-renderer';
import PropComponent from './PropComponent';

//snapshot testing
//allows us to take a snapshot of a component and test against
it('renders properly', () => {
    //can tell the renderer what component/ componenets we want it to create
    //when we first run it, it wil lautomatically make a snapshot of this
    const component = renderer.create(
        <PropComponent name="TestName" age={21}>
            <p>Test Text</p>
        </PropComponent>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

//run tests with npm test. check the README for more info on commands
//runs in "watch mode", so everytime you save while in that mode it just reruns your tests