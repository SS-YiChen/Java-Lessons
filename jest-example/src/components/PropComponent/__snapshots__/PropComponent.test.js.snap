// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
Array [
  <h1>
    Name: 
    TestName
  </h1>,
  <h2>
    Age: 
    21
  </h2>,
  <p>
    Test Text
  </p>,
]
`;
